experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+"""Heatmaps and reward plotting code for debugging MountainCar."""
+
+from functools import partial
+import os
+from pathlib import Path
+import pickle
+from typing import Dict, List, Optional, Sequence, Union
+
+import gym
+from matplotlib import pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from stable_baselines.common.vec_env import DummyVecEnv, VecNormalize
+
+from imitation import util
+from imitation.policies.serialize import load_policy
+from imitation.rewards.serialize import load_reward
+from imitation.util.reward_wrapper import RewardFn
+
+MC_POS_MIN, MC_POS_MAX = -1.2, 0.6
+MC_VEL_MIN, MC_VEL_MAX = -0.07, 0.07
+MC_GOAL_POS = 0.5
+
+
+# Utility for calculating the next observation s'.
+# Required for evaluating AIRL reward at arbitrary (s, a) points.
+def make_next_obs(obs, acts) -> np.ndarray:
+  """Utility for calculating the next observation s'.
+
+  Required for evaluating AIRL reward at arbitrary (s, a) points.
+  """
+  env = gym.make("MountainCar-v0")
+  obs = np.array(obs)
+  acts = np.array(acts)
+  next_obs = []
+  for ob, act in zip(obs, acts):
+    assert ob.shape == (2,)
+    env.reset()
+    env.env.state = ob
+    next_ob, *_ = env.step(act)
```

Comment:
```suggestion
    next_ob = env.step(act)[0]
```

Don't have strong feelings on this, the current method is more visible (`[0]` is easily missed), but it does seem a bit indirect.